src/main/java/com/gojek/beast/converter/RowMapper.java

```diff
@@ -34,22 +35,40 @@ public RowMapper(ColumnMapping mappings) {
         if (mapping == null) {
             throw new ConfigurationException("BQ_PROTO_COLUMN_MAPPING is not configured");
         }
+
+        List<DynamicMessage> messageWithUnknownFields = getMessageWithUnknownFields(getDynamicMessageFields(message));
+        if (messageWithUnknownFields.size() > 0) {
+            statsClient.count("kafka.error.records.count,type=unknownfields," + statsClient.getBqTags(), 1);
+            if (failOnUnknownFields) {
+                throw new UnknownProtoFieldFoundException(message.toString());
+            }
+        }
+
         return getMappings(message, mapping);
     }
 
+    private List<DynamicMessage> getDynamicMessageFields(DynamicMessage message) {
```

Comment:
As discussed over call, using an accumulator will help in collecting the messages properly
`
        final List<DynamicMessage> allNestedMessagesInProto = getAllNestedMessages(message, List.of(message));
        List<DynamicMessage> messageWithUnknownFields = getMessageWithUnknownFields(allNestedMessagesInProto);
        if (messageWithUnknownFields.size() > 0) {
            statsClient.count("kafka.error.records.count,type=unknownfields," + statsClient.getBqTags(), 1);
            if (failOnUnknownFields) {
                throw new UnknownProtoFieldFoundException(message.toString());
            }
        }
        return getMappings(message, mapping);
    }

    private List<DynamicMessage> getAllNestedMessages(DynamicMessage message, List<DynamicMessage> nestedMessages) {
        List<DynamicMessage> children = message.getAllFields().values().stream()
                .filter(v -> v instanceof DynamicMessage)
                .map(o -> (DynamicMessage) o)
                .collect(Collectors.toList());
        nestedMessages.addAll(children);
        children.forEach(m -> getAllNestedMessages(m, nestedMessages));
        return nestedMessages;
    }`